pkg/client/client.go

```diff
@@ -1006,6 +1017,12 @@ func (c *Client) WaitForAlertmanager(ctx context.Context, a *monv1.Alertmanager)
 		if ctx.Err() != nil && lastErr != nil {
 			err = lastErr
 		}
+		s, _ := c.kclient.AppsV1().StatefulSets(a.GetNamespace()).Get(ctx, "alertmanager-main", metav1.GetOptions{})
+		generationAttemps := s.Status.ObservedGeneration - lastGeneration
+		if generationAttemps > 5 && s.Status.ReadyReplicas != s.Status.Replicas {
```

Comment:
I would leave a comment here explaining the reasoning about the 5 attempts.

Reply:
done